compiler/semmagic.nim

```diff
@@ -656,10 +656,6 @@ proc magicsAfterOverloadResolution(c: PContext, n: PNode,
     if not checkIsolate(n[1]):
       localError(c.config, n.info, "expression cannot be isolated: " & $n[1])
     result = n
-  of mPred:
```

Comment:
why remove it?

Reply:
Because it's wrong for the JS backend. The second argument is always an `int` (`Number` in JS), but the first argument can be a 64 bit int (`BigInt` in JS by default) and you can't add a `BigInt` and a `Number` with `+`.

Reply:
need fix in jsgen, semmagic is frontend.

Reply:
What's the point of this transformation anyway? And why is only `mPred` transformed (and not `mSucc`, `mInc` and `mDec`)? This only complicated jsgen for no good reason and makes it more confusing (it did confuse the hell out of me when trying to find the source of the bug, because I didn't expect `mPred` to be replaced).

Reply:
mPred is supset of sub int and sub uint AFAIK.it's specially handled in different ways in backends.